train_dynamics.py

```diff
@@ -213,17 +216,13 @@ def train_step(state, inputs):
         args.batch_size,
         *image_shape,
     )
+    dataloader = (jax.make_array_from_process_local_data(videos_sharding, elem) for elem in dataloader) # type: ignore
```

Comment:
Nice one. This applies the function lazily and creates a generator. If you were to use these brackets: [], it would load the entire dataset in memory.